hbase_sql/src/main/java/com/splicemachine/mrio/api/core/AbstractSMInputFormat.java

```diff
@@ -103,7 +103,22 @@
                 }
                 SubregionSplitter splitter = new HBaseSubregionSplitter();
 
-                return splitter.getSubSplits(table, splits, s.getStartRow(), s.getStopRow());
+                List<InputSplit> lss= splitter.getSubSplits(table, splits, s.getStartRow(), s.getStopRow());
+                //check if split count changed in-between
+                List<Partition>  newSplits = clientPartition.subPartitions(s.getStartRow(), s.getStopRow(), refresh);
```

Comment:
If you do it this way you need to set refresh to true on the second call, otherwise you might pull the region info from the cache and miss the split. You were lucky while testing because the compaction caused by the split refreshes the cache, but that happens on a different thread and there's no guarantee that will always happen at the right time.